### DISCUSSION TITLE
[Unstable] PVP "rework" is wrong

### DISCUSSION THREAD

**Discussion by Scrymmery** [FakeFishGames/Barotrauma#14236] [Unstable feedback]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

1. First of all, diving suits not spawning on subs anymore. It's obvious it shouldn't be like that, because there are also no materials to craft some. **How should we PVP then?** Just shooting eachother from the cannons? It's boring, bring back the suits pls.
2. Gameplay settings. For some reason they're set to 200% for Win score, and Stun resistance is also 50%. I suppose it's there only for testing reasons, but ig it's obvious it should be 100% and 0%. Like wtf, why **default(100% according to math)** modifier is **200%**? Same with the stun, default settings should give default, unmodified experience, why should i turn off stun resistance and change win score modifier back to 100% if i want to play vanilla pvp? Please, set it to default ones. _(Also, kinda strange that it's stun RESISTANCE modifier. Would be much better to standardize both slider as modifiers. Like amount of applied stun which is 100% by default, but can be more than 100 or 0)_
3. Not a bug, but a logical suggestion. If there are gameplay modifiers, why there are no settings for _healing items cooldown_, _modifiers for healing amount(like, if i use morphine and it will heal me twice more effectievly)_ and _amount of applied damage of other types_? It's really **confusing**, why am i able to change _only_ these two aspects of PVP? As soon as you can modify amount of applied stun, i believe it's very easy to implement same thing for others, giving us **COMPLETE** list of modifiers, not only these two.
4. No preference players not able to vote for a sub. Why they are forced to take a team to vote for a sub? Their votes just should be added to both teams, or only one if opposite team didn't voted for that sub. So, if there is 4/4 azimuth and 4/4 dugong, and neutral players voted for azimuth, it means more players want to play on azimuth, and it will be 5/5 azimuth and 4/4 dugong. Makes **much more** sense.
5. And again, second suggestion is to make old way of voting for the sub toggleable, so you can switch between "fair" voting where all server vote for sub that will be used for both teams, and new "unfair" voting where each team is allowed to pick their sub. Like, what should we do if we play full randomized teams? 

I'm playing PVP for a very long time, and i know how PVP works now, and in this state, looks like when you release update, PVP will be a huge mess, so please, fix it. I know you trying to make players experience better, but:

- lack of diving suits will just make hosts to add them as additional cargo
- undefault settings will make it to switch "back" for "default"
- a large amount of swearing due to unlogical ablity to change stun amount, but not burn damage for example
- also a large amount of swearing because it's almost impossible to play full randomized teams as it was before

PVP is my favorite game mode so far, and i really happy it finally getting an update after being abandoned for so long, but i also don't want you to ruin balance or even ruining entire game mode. Additions would be really cool, just add new stuff, and don't mess with what PVP players got used to by ages. I believe it wouldn't be hard to fix what i mentioned before, but these small fixes will make PVP **MUCH BETTER** than it could be if you leave things as the are.

### Reproduction steps

1. Start a game.
2. Start PVP server.
3. Try to PVP.
4. Sadness((

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.6.100.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by LePiromano** [comment]
1st is just a  visual bug, there's suits in the lockers you just cant see them from the outside

**Comment by Regalis11** [comment]
Thank you for the report! I would like to stress that the PvP mode rework is still very much a work in progress - what you see in the latest unstable build is some early "groundwork" for the PvP mode and subject to change.

> First of all, diving suits not spawning on subs anymore. It's obvious it shouldn't be like that, because there are also no materials to craft some. How should we PVP then? Just shooting eachother from the cannons? It's boring, bring back the suits pls.

They have not been removed - this is a bug in the current unstable build that prevents the sprites from appearing on the lockers.

> Gameplay settings. For some reason they're set to 200% for Win score, and Stun resistance is also 50%. I suppose it's there only for testing reasons, but ig it's obvious it should be 100% and 0%. Like wtf, why default(100% according to math) modifier is 200%? Same with the stun, default settings should give default, unmodified experience, why should i turn off stun resistance and change win score modifier back to 100% if i want to play vanilla pvp? Please, set it to default ones. (Also, kinda strange that it's stun RESISTANCE modifier. Would be much better to standardize both slider as modifiers. Like amount of applied stun which is 100% by default, but can be more than 100 or 0)

I think you've misunderstood the "win score" slider. It's not a multiplier/modifier: it's simply the score the team needs to reach to win. What the "score" represents depends on the game mode: in the deathmatch mode you score points from kills, in the king of the hill mode you score by controlling a beacon station, and in future (or modded) game modes there might be other ways to score points as well.

> Not a bug, but a logical suggestion. If there are gameplay modifiers, why there are no settings for healing items cooldown, modifiers for healing amount(like, if i use morphine and it will heal me twice more effectievly) and amount of applied damage of other types? It's really confusing, why am i able to change only these two aspects of PVP? As soon as you can modify amount of applied stun, i believe it's very easy to implement same thing for others, giving us COMPLETE list of modifiers, not only these two.

The reason why we added stun modifiers first was that we felt stun weapons are so overpowered in PvP. A modifier that allows reducing stun durations has been very often requested by players too. That's not to say we couldn't add more multipliers though! A modifier for heals could be useful, although it's not quite as simple as stuns, because there's so many different afflictions that behave in such a different way that a simple multiplier on healing items (with side-effects, delays, durations and such) would not cut it to linearly scale their effectiveness.

> No preference players not able to vote for a sub. Why they are forced to take a team to vote for a sub? Their votes just should be added to both teams, or only one if opposite team didn't voted for that sub. So, if there is 4/4 azimuth and 4/4 dugong, and neutral players voted for azimuth, it means more players want to play on azimuth, and it will be 5/5 azimuth and 4/4 dugong. Makes much more sense.
> And again, second suggestion is to make old way of voting for the sub toggleable, so you can switch between "fair" voting where all server vote for sub that will be used for both teams, and new "unfair" voting where each team is allowed to pick their sub. Like, what should we do if we play full randomized teams?

I completely agree, there should be a way to vote for the sub regardless if you want to choose a team or not.

**Reply by JeroonBosch** [reply]
To add upon this:

- Slider values are currently wrong. 0% stun resistance and 100% win score are the intended default values.
- Stuns can feel bad, especially in PvP, as the affected player loses control of their character for the duration of a stun. In the hands of player, a stun means the ability to remove someone's diving suit as well as their weapons while they are stunned, making them very powerful. Adding a slider for stuns specifically therefore can be very impactful and drastically change what a PvP match looks like.
- While we may expand on the list of modifiers, a complete list of modifiers is unlikely to much beyond making there be too many options in the UI. If you've got any good arguments for certain modifiers we've missed, we're happy to look into those though.

**Reply by Scrymmery** [reply]
> * While we may expand on the list of modifiers, a complete list of modifiers is unlikely to much beyond making there be too many options in the UI. If you've got any good arguments for certain modifiers we've missed, we're happy to look into those though.

Why then not create a separate tab for modifiers? Like a list of additional cargo, where you will have a bunch of toggles and slider to configure gameplay as you want (something like the minecraft 1.8 customized world menu :skull:)
Medicine usage delay should be modifiable ig, it affects PVP not less than stun, because when there was no cooldown, you were able to use as much medicine as possible untill you overdose. You were supposed to have weapon that make your enemy ran out of medicine, and then another weapon to finish him. 
Also, toggleable skill threshold is a good idea, so you won't suffer bc your skills are low.

PVP always was (atleast, in RU segment of barotrauma community) highly skill-dependent gamemode, and i can state that stuns isn't bad. Stun is bad if you're a noob, but in that state every weapon used against you could be considered as bad. Because without stun winner is determined by DPS, and barotrauma is very bad in balance, that's why entire PVP community hated almost every update. Like, being unable to stun your enemy, even if you outsmart him and win a bit of time, anyway the winner will be the one with explosive autoshotgun and not with SMG. Ofc, it depends on host if there is perk loot, but before halloween update there were **no option to ruin balance**, bc game was balanced for PVP well enough. **So if you're looking for an ability to make perfectly balanced PVP - look towards pre-NewFrontiers versions of game**.

**Reply by Regalis11** [reply]
Thank you for the follow-up, this sort of feedback is valuable to us in getting the PvP rework to a releasable state. :) I just have a couple of questions:

> before halloween update there were no option to ruin balance

Which changes in the halloween update do you mean specifically? I'm also not sure I understand why you consider the _option_ to reduce stuns a bad thing. I understand that many players prefer the longer stuns, and that's why we made the multiplier an adjustable setting - why's it an issue if it's an entirely optional thing?

**Reply by NilanthAnimosus** [reply]
Hm, regarding the awkwardness of afflictions and sliders for "Healing" purposes, it could be possible if you don't go by items, but instead the affliction's rate of change when decreasing - then base it off the Type of affliction (we already mark them as "Damage" "Burn" "Bleeding" "Bloodloss" "alieninfection" "poison" "Debuff" "paralysis" "oxygenlow" etc - you could even expand the types available, and what types to change by slider. Instead of trying to go item by item it might make more sense to simply act as multipliers on positive or negative changes to afflictions themselves.

If some fall under weird categories like debuffs they could also have additional types or deritive types, some marker added like tags or an XML true/false, or other means to specify what should and shouldn't be effected by sliders. It may not always be balanced, but it is then a choice of the server owner and no different than if they just went out to edit all the item XMLs, but without requiring to modify every mods values. Its probably simpler than trying to create some kind of item-specific multiplier system at least.

**Reply by NilanthAnimosus** [reply]
Only main issue with sliders that change game mechanics (wanted or not) is items not behaving as expected in a game one joins without knowing and seeing said settings clearly. Like finding one server morphine heals half, and on another an opium heals 90% of the vitality bar.

As a note - If I had my way I would have like 100 sliders and 500 values in most XML like a crazy person for everything one could tweak, Im not against it, its just Im not sure how visible the game settings are (unlike server settings) to curious players that will very drastically effect gameplay mechanics in this case.

**Reply by Scrymmery** [reply]
> why you consider the option to reduce stuns a bad thing

I'm not really against of option to reduce stuns, just my opinion on why stuns isn't bad. But talking about it, after update the number of PVP servers will increase for a short time, and i really doubt they'll be good at PVP, possibly whiners who'll ban you bc you more skilled than the host lol. Not a big issue tbh. So, stun modifier is good, tho it'd be better if it's not stun resistance but amount of recieved stun as i mentioned before.

> Which changes in the halloween update do you mean specifically?

The main change that affected balance in halloween update is toolbox removal. But halloween update is just a reference point. It's staying between new frontiers and uncharted depths update. New frontiers changed alot, and didn't affected balance alot. And starting from uncharted depths, following updates balance was mainly progress-dependent, and made game more casual. It could be ok for campaign mode, but i sure you weren't taking PVP balance into account when developing those updates (even removal of o2 tanks exploding other tanks changed alot in PVP). That's why halloween update is used(atleast, across old RU players) as a mid-point between rise and recession of PVP.
(also, some of players still haven't forgiven you for ITA, Thalassophobia, EK and other mods the situation with which began after halloween update)

So, if you look past halloween update, there'll be a lot of features that were used commonly in PVP, for example:
- Piercing railgun shells
- Instastun
- Steroids
- Chain reaction explosions for o2/welding tanks
- No closing menus on microstuns
- No medicine cooldowns
- Faster diving suits

Ofc these are looks like OP ones, but that's what was making PVP balanced, when you could get killed pretty fast and easily, you have the same options. And methods for fast killing not quite simple comparing to newer versions with slower tempo of PVP, where you usually kill your enemy without any tactics holding one button.

**Reply by Scrymmery** [reply]
> Only main issue with sliders that change game mechanics (wanted or not) is items not behaving as expected in a game one joins without knowing and seeing said settings clearly. Like finding one server morphine heals half, and on another an opium heals 90% of the vitality bar.

Would be cool to show it in ingame item tooltips, like "Morphine (110% Effectiveness)", and make modifiers list accessible to view for everyone in lobby

**Reply by Regalis11** [reply]
> Hm, regarding the awkwardness of afflictions and sliders for "Healing" purposes, it could be possible if you don't go by items, but instead the affliction's rate of change when decreasing

@NilanthAnimosus I don't think this would make much difference in practice, because most afflictions don't decrease by themselves (or if they do, it's very slow). For example, the rate of change of the normal "damage" afflictions is 0, so a multiplier on that would do nothing.

> if it's not stun resistance but amount of recieved stun as i mentioned before.

@Scrymmery I don't quite understand what you mean by this. I see "resistance" as a reduction of the amount of received stun (which is what the slider affects currently) - how would you prefer it to work instead?

> also, some of players still haven't forgiven you for ITA, Thalassophobia, EK and other mods the situation with which began after halloween update

Hmm, what situation do you mean? If you're referring to updates during the Early Access period frequently forcing modders to make changes to their mods, I understand it was very frustrating to modders and to the players using those mods, but I would also argue many of those changes were necessary: many underlying systems in the game needed to be improved (both technically and design-wise) to allow for the further development of the game, and an unfortunate side-effect of that was that mods would frequently become incompatible. Which was kind of to be expected and something we warned modders about: the game was still very much a work in progress and almost everything was still subject to change, so a modder doing expansive changes or additions to the vanilla content and gameplay would be taking a conscious risk that their work will become invalidated by some of the changes to the vanilla game.

**Reply by Scrymmery** [reply]
> @Scrymmery I don't quite understand what you mean by this. I see "resistance" as a reduction of the amount of received stun (which is what the slider affects currently) - how would you prefer it to work instead?

literaly amount of stun recieved. For now, you can set 80% of **stun resistance**, which means you'll recieve 80% less stun. The problem is that you can't set resistance to negative values (you can actually, but it won't look good). Instead of **stun resistance**, i suggest **stun modifier**, so if you set it to 80%, it means you'll recieve only 80% of stun, and if you set it to zero it will be the same as 100% of **stun resistance**. The main thing is that you can set it to 300%, so stuns will be three times more effective.
So, if simplify, formula for _stun resistance_ is `STUN_RECIEVED=INCOMING_STUN*(1-MODIFIER)`
And for _stun modifier_ is `STUN_RECIEVED=INCOMING_STUN*MODIFIER`


> Hmm, what situation do you mean?

Not compatibility issues, i mean that mod content was stolen and added into base game, while modders got nothing from this. Ofc it can be referenced as inspiration, but it doesn't change a lot. For example, when Thalassophiba was copied, the original mod became incompatible with the game, and became unnecessary as Thalassophobia content was implemented in original game. Authors didn't got anything except for stolen ideas and broken mods. Not even a mention in game credits(some are the mentioned, but only because of they were contributing on github). Anyways, that does not apply to our theme of PVP/balance.

**Reply by ubertpendragon** [reply]
Not sure why anyone would set stuns to 300%

PvP always had an issue with say stun grenade launcher being effective instakills so it makes sense that you'd be able to tone it down.


Not a fan of Slider/option creep that being said, the server menu looks more technical than the actual sub management game itself and pub games have inconsistent rulesets across the board

**Reply by Regalis11** [reply]
> For now, you can set 80% of stun resistance, which means you'll recieve 80% less stun. The problem is that you can't set resistance to negative values (you can actually, but it won't look good)

Ah, now I see what you mean, thank you for the clarification! Yeah, I agree a multiplier/modifier on the stun could be more useful than the resistance setting.

> Not compatibility issues, i mean that mod content was stolen and added into base game, while modders got nothing from this

This is a pretty severe accusation, and simply not true. We have **not** stolen any content from these mods to the main game. If you can point out any examples of stolen content, please do, because I would like to make sure that content gets removed and/or that we get a permission from the modders to use it.

Taking inspiration (as in ideas, as opposed to actual content) from mods is another matter, and that we have done to some extent - however, less than you probably think. Many things we've added to the game during the Early Access were not ideas we got from mods, but things that had been requested by the community for ages, and/or things that we had planned to implement for a long time and that were clearly still missing from the game. The modders just noticed many of the same shortcomings as us, and managed to address them before we did. For example, the idea of the genetics system didn't come from Thalassophobia, but SS13, and to my knowledge the designer who designed it hasn't even tried Thalassophobia. And things ITA adds are mainly somewhat generic (but very good!) additions to make the content in the game more varied and to balance the difficulty - and those were clearly things we needed to do in the vanilla game too even if modders got to it before we did. I think the same applies to the EK mods - they added some severely needed variety to the game's weapons, and did so by adding such a variety of new weapons that it would've been very difficult to avoid some overlap between the vanilla additions and the EK mods.

> Not even a mention in game credits(some are the mentioned, but only because of they were contributing on github)

Which of them are not mentioned? We have wanted to mention the developers of the most popular mods in the "community actives" section of the credits (although we've always asked first, and some of them haven't wanted their name included), but it could be we have missed some of them. On a cursory look, it seems to me that the creators of ITA, Thalassophobia and EK are all included.

**Reply by JeroonBosch** [reply]
> Medicine usage delay should be modifiable ig, it affects PVP not less than stun, because when there was no cooldown, you were able to use as much medicine as possible untill you overdose. You were supposed to have weapon that make your enemy ran out of medicine, and then another weapon to finish him.

To me personally this does not sound like very interesting or good gameplay though.

> Ofc these are looks like OP ones, but that's what was making PVP balanced, when you could get killed pretty fast and easily, you have the same options.

This definition of balance, where some options are overpowered and can instakill, is not something I agree with. Just because both teams have access to it doesn't imply balance. Then it becomes more of a knowledge game, knowing which options are OP and how to (ab)use game mechanics to remove the ability to react from other players.

**Reply by Scrymmery** [reply]
> We have **not** stolen any content from these mods to the main game.

I'm not the lawyer, but ig that legally speaking, you're right. But since the main content of mods is ideas, stealing content is pretty similiar to stealing ideas. It's obvious that there's no copied XMLs from mod, but mechanics are pretty same.

> The idea of the genetics system didn't come from Thalassophobia, but SS13, and to my knowledge the designer who designed it hasn't even tried Thalassophobia. 

Yes, both mod and game took inspiration on SS13, but iirc vanilla genes implementation are closer to Thalassophobia implementation, than SS13. Sadly i couldn't find older versions of Thalassophobia to compare. 

> On a cursory look, it seems to me that the creators of ITA, Thalassophobia and EK are all included.

Oh, they're included, my bad. For some reason i remember that they weren't. 

I don't accuse you for stealing the mods, as i said, _there are still players who haven't forgiven you for that_(and that's **just another reason** why halloween update is so important on timeline). If mods really gave their permission to implement ideas from mods, then it's ok. But on release of those updates, they weren't mentioned in changelogs(like special thanks), there also were no news about that except that mods are abandoned. Also one of the problems is that before halloween update, content added in updates was either community suggestions or your own ideas, and then you just started implementing mods. From the side of average players, it really looked like devs are out of ideas and taking modders ideas.
So, if there's a permission from modders, i apologize for calling it a steal.

**Reply by Scrymmery** [reply]
@JeroonBosch 
> Just because both teams have access to it doesn't imply balance.

yes, you're right. The thing is that before OP guns were **actually** balancing eachother, a bit hard to explain, but for example, if both players had tasers(which are considered OP nowadays), then it's matter of how to properly engage the fight to taser your opponent and not die(Ballistics and timings are does a lot of impact). And compare to assault riffle for example, no matter what you do, the first one who pressed the button will win. That's the difference. Previous balance worked well because two OP weapons were nerfing eachother, but now two OP weapons are still OP.

>Then it becomes more of a knowledge game, knowing which options are OP and how to (ab)use game mechanics to remove the ability to react from other players.
Yes, that's what makes baro PVP different from shitty shooters. You were supposed to know how game works, what weapon to pick against others, have to look for enemies, you're able to mess with enemy wirings, etc. That's about possibilities, not just reaction and aim. And game abuses is what makes game skill-dependent. Just take a look at ultrakill: you can choose to not abuse game mechaincs and play below average, but dev implemented abuses as a features, so if you'd like to complete skillchecks and challenges, you just have to abuse the game, and it's not that easy.

**Reply by Regalis11** [reply]
I think you're ignoring my points about the many of the ideas in the mods being features that have been requested by the community for years before early access, and also kind of "obvious" additions that were at the time missing from the game. We did get some ideas from mods, but many of the ideas were also things that we had planned for a long time, or us coincidentally ending up with similar designs as in mods (which is not surprising - it's natural that the two separately designed genetic systems for example are similar, because they're both designed to "fill the same gap" in the same game and leverage the same existing mechanics and systems). 

I can see how it may have come across as us "running out of ideas and taking modders ideas", but in reality it was more of a case of us starting to "fill in the gaps" in the game and add more content where it was needed, which modders had already been doing since the Early Access release - and we often ended up going for similar solutions as modders. I think that if we couldn't use the same ideas that others have used, it'd be extremely difficult to add anything to the game because of the size of the modding community and the sheer number of different kinds of mods - and actually, many of the ideas used in those mods had also been brought up tons of times in the Discord, Github and Steam discussions, so I suppose you could also see it as both us and the modders "stealing" from the pool of community ideas that we've gathered along the years. :D

**Reply by Regalis11** [reply]
> yes, you're right. The thing is that before OP guns were actually balancing eachother, a bit hard to explain, but for example, if both players had tasers(which are considered OP nowadays), then it's matter of how to properly engage the fight to taser your opponent and not die(Ballistics and timings are does a lot of impact). And compare to assault riffle for example, no matter what you do, the first one who pressed the button will win. 

I get what you mean, but I don't think that's necessarily a good way to go about balancing. If there are OP guns, you're essentially forced to use those OP guns and the less strong ones become obsolete. For example, if stuns are the easiest way to kill someone, that becomes the only viable tactic in the game.

And yes, you raise a good point about assault rifles. They are very strong, and that's the reason they're locked behind talent progression, has limited ammo and has the serious drawback that you can't put in in your normal inventory slots. To my knowledge, it shouldn't even be possible to get them in the PvP mode though unless you're playing on a custom sub pre-supplied with those or spawning them as extra cargo? And I would argue that it's still not a matter of who presses the button first - it's a matter of who ends up at the wrong side of the assault rifle's barrel, and you can take measures to not put yourself in that situation.

**Reply by NilanthAnimosus** [reply]
I was thinking in regards to affliction rate of change the applied rate from status effects and not the natural decrease some afflictions (IE. burn or bloodloss) have. that is if something is set to change it by -1, instead change it by -1.5 if its at a 50% increased multiplier. That way something set to heal away 18 damage could instead heal away 27 damage or such, without going off strictly the items, just the afflictions and what is modifying them. (so items, beds, natural change, all might be effected by a general modifier).

Im sure theres probably some reason not to do it that way, and in the end most balancing could be applied by a mod too, but as far as a slider option goes I feel going by affliction would be better than per-item basis for simplicity is all. Though I think an obvious drawback is it'd effect anything (like none humans) and not just item based healing.

**Comment by Scrymmery** [comment]
>you're essentially forced to use those OP guns and the less strong ones become obsolete.

Yes, but since there were less weapons, all of them was usable. Stun is good if you're close enough, but SMG was able to deal enough damage to drain all the medicine, or deny you from stunning. Revolvers were good at long distances, so you can deal alot of bleeding without risk of getting stunned. Explosions are good if you want to kill enemy that out of your sight. All the weapons were useful, and you always had a lot of options to kill an enemy.

>To my knowledge, it shouldn't even be possible to get them in the PvP mode.

Nobody ever cared of it lol, as you said there is additional cargo and custom subs, so it's obvious perk loot will be used in pvp. It also didn't takes much effort to aim, so if one player is in range, then second player too. And once you're shot even once, you're probably dead. So it's defiently not a matter of skill or game knowledge. That's actually the situation where OP weapons makes most of other weapons useless.

**Reply by Regalis11** [reply]
Well tbh if you're choosing to use later-game weapons in the PvP mode (even though they're normally not available there), it's to be expected that the balance will be off. There are weapons in the game that are objectively better than others, but won't become available until later in the game, and in my opinion that's the way it has to be for there to be some sense of progression in the weaponry in the campaign mode. If you're opting to use these later-game weapons and gear in the PvP mode, it seems like a self-imposed problem to me.

**Reply by Scrymmery** [reply]
Yes, it's self-imposed problem, and usually if the host adds a lot of perk loot, host gets blamed by skilled players. The point is that in past you were able to use all the game content and it was hard to ruin balance. So if the goal to make balanced PVP without ruining campaign progression, there should be settings that will allow to balance those. More flexible settings for PVP the better PVP will be(ideally if it'll be possible to configure balance so it'll be a pre-release balance). Also will bring much more players in game mode.

**Reply by Regalis11** [reply]
That is true yeah, hosts can screw things up by choosing the "wrong" kind of extra cargo or subs for the PvP mode. However, I would argue that they should be allowed to: if someone wants to play PvP with late-game weapons (making "normal" gear useless), they should be allowed to do so.

>  there should be settings that will allow to balance those

The way I see it, this setting already exists in a sense: it's the extra cargo and submarine selection. If you don't want to play with late-game gear or subs, don't choose those. There is going to be content in the game that has been designed for later stages of the campaign, and I don't really see much value in attempting to "nerf" that content specifically for the PvP mode, especially when it's normally never available there.

**Reply by Scrymmery** [reply]
>If you don't want to play with late-game gear or subs, don't choose those.

Then what if someone wants to play with fissile accelerator, but doesn't want to make others weapons useless? Again, i suggest not to make the best balance, but to give players an ability to fully configure balance however they want.

**Reply by Regalis11** [reply]
> give players an ability to fully configure balance however they want.

I feel like this isn't as simple as it sounds. That would require adding an enormous amount of configurable options: ways to configure damage per-weapon and/or per ammo type, ways to adjust the damage from ammo depending on which weapon it's fired from, ways to configure reload times, spread and such, ways to add drawbacks to weapons to compensate for their high DPS... There's also the fact that balancing is hard, it takes a lot of fine-tuning and testing to get the weapons to be even roughly in balance, and I doubt that many players would be willing to go through that effort. 

The way I see it, it might even be simpler just to dive into the weapon XMLs, do the balancing there and publish the changes as mod than to do it via in-game adjustable settings.

**Reply by Scrymmery** [reply]
I doesn't know barotrauma code that much, but it should be possible to fetch all weapons from xml and their base damages, isn't it?. According to my programmer experience, problems like these aren't that hard to solve.
But if for some reason it's hard to do, atleast some easier modifiers would be enough. Like base hp amount, amount of incoming damages, etc. You got what i mean. A separate tab for modifiers is good to not overload server UI

**Reply by LePiromano** [reply]
it is possible but adding a stat slider for every single item just for one game mode feels like an exaggeration, if you want to make that many tweaks for your own play session it's better to just make a mod

**Reply by Regalis11** [reply]
> I doesn't know barotrauma code that much, but it should be possible to fetch all weapons from xml and their base damages, isn't it?

Yes, we could just add a multiplier to the damages, but my point was that there's more to balancing weapons than just the raw DPS. There's tons of different factors that affect the balance of the weapons, and to balance weapons "properly" you may need to consider and adjust all of those. I agree with LePiromano, if you wish to make the late-game weapons balanced with the basic weapons, I feel like creating a mod would be your best bet.
